jni/src/faiss_wrapper.cpp

```diff
@@ -223,28 +223,16 @@ jobjectArray knn_jni::faiss_wrapper::QueryIndex_WithFilter(knn_jni::JNIUtilInter
     omp_set_num_threads(1);
     // create the filterSearch params if the filterIdsJ is not a null pointer
     if(filterIdsJ != nullptr) {
-        int *filteredIdsArray = jniUtil->GetIntArrayElements(env, filterIdsJ, nullptr);
+        int64_t *filteredIdsArray = jniUtil->GetLongArrayElements(env, filterIdsJ, nullptr);
         int filterIdsLength = env->GetArrayLength(filterIdsJ);
         std::unique_ptr<faiss::IDSelector> idSelector;
-        FilterIdsSelectorType idSelectorType = getIdSelectorType(filteredIdsArray, filterIdsLength);
-        // start with empty vectors for 2 different types of empty Selectors. We need define them here to avoid copying of data
-        // during the returns. We could have used pass by reference, but we choose pointers. Returning reference to local
-        // vector is also an option which can be efficient than copying during returns but it requires upto date C++ compilers.
-        // To avoid all those confusions, its better to work with pointers here. Ref: https://cplusplus.com/forum/general/56177/
-        std::vector<faiss::idx_t> convertedIds;
-        std::vector<uint8_t> bitmap;
-        // Choose a selector which suits best
-        if(idSelectorType == BATCH) {
-            convertedIds.resize(filterIdsLength);
-            convertFilterIdsToFaissIdType(filteredIdsArray, filterIdsLength, convertedIds.data());
-            idSelector.reset(new faiss::IDSelectorBatch(convertedIds.size(), convertedIds.data()));
+        if(filterIdsTypeJ == BITMAP) {
+            const int bitsetArraySize = filterIdsLength * 8;
+            uint8_t *bitmap = reinterpret_cast<uint8_t*>(filteredIdsArray);
```

Comment:
Few more comments.
1. It is not filteredIdsArray but filteredIdsBitmap. We need to name the variable more properly.
2. I wonder this conversion is correct or not. For example, 11 in int64_t in forms of binary representation is 
```
0000000000000000000000000000000000000000000000000000000000000001 (64bit)
0000000000000000000000000000000000000000000000000000000000000001
```
in uint8_t it should be
```
00000001 (8bit)
00000000
00000000
00000000
00000000
00000000
00000000
00000000
00000001
00000000
00000000
00000000
00000000
00000000
00000000
00000000
```

Reply:
@heemin32 i know your concern about the safety and consistency for the type changed `uint8_t *bitmap = reinterpret_cast<uint8_t*>(filteredIdsArray);`

so i create a new `IDSelector#IDSelectorJlongBitmap` in new commits, for the bitmap using jlong ,and`is_member` logic reuse `Lucene#FixedBitSet`

https://github.com/opensearch-project/k-NN/blob/bfbfb55ab8eaedee1e19894ab5001e3339e9c00a/jni/src/faiss_wrapper.cpp#L34-L57